test/runtime/adapter-config.spec.js

```diff
@@ -0,0 +1,139 @@
+            var outputConf = result.sinks.table;
+            expect(outputConf[0]).to.deep.equal(inputConf);
+        });
+    });
+    
+    it('load array', function() {
+        var inputConf = [
+            {
+                id: 'first',
+                host: 'testHost1'
+            },
+            {
+                id: 'second',
+                host: 'testHost2'
+            }
+        ];
+        var name = inputConf[0].host;
+        configure(name, inputConf);
+        
+        return check_juttle({
+            program: `read ${name}`
+        })
+        .then(function(result) {
+            expect(result.sinks.table).to.have.length(inputConf.length);
+            
+            var outputConf = result.sinks.table;
+            expect(outputConf).to.deep.equal(inputConf);
+        });
+    });
+    
+    it('load conf under instances key', function() {
+        var inputConf = { 
+            instances: [
+                {
+                    id: 'first',
+                    host: 'testHost3'
+                },
+                {
+                    id: 'second',
+                    host: 'testHost4'
+                }
+            ] 
+        };
+        var name = inputConf.instances[0].host;
+        configure(name, inputConf);
+        
+        return check_juttle({
+            program: `read ${name}`
+        })
+        .then(function(result) {
+            expect(result.sinks.table).to.have.length(inputConf.instances.length);
+            
+            var outputConf = result.sinks.table;
+            expect(outputConf).to.deep.equal(inputConf.instances);
+        });
+    });
+    
+    it('err missing id', function() {
+        var inputConf = [
+            {
+                host: 'testHost5'
+            },
+            {
+                id: 'second',
+                host: 'testHost6'
+            }
+        ];
+        var name = inputConf[0].host;
+        configure(name, inputConf);
+        
+        return check_juttle({
+            program: `read ${name}`
+        })
+        .then(function(result) {
+            throw new Error('unexpected success');
+        })
+        .catch(function(err) {
+            expect(err.message).equal(`One or more configuation instances for adapter ${name} is missing an id.`);
+        });
+    });
+    
+    it('err instances not array', function() {
+        var inputConf = { 
+            instances: {
+                id: 'first',
+                host: 'testHost7'
+            },
+            host: 'hostPlaceholder1'
+        };
+        
+        var name = inputConf.host;
+        configure(name, inputConf);
+        
+        return check_juttle({
+            program: `read ${name}`
+        })
+        .then(function(result) {
+            throw new Error('unexpected success');
+        })
+        .catch(function(err) {
+            expect(err.message).equal(`instances key in configuation for adapter ${name} must be an array.`);
+        });
+    });
+});
```

Comment:
I think it'd also be good to have some tests to verify what happens when `path` points to a nonexistent directory or an otherwise invalid path, coupled with the change above.